### PR TITLE
fix: Return 404 when deleting non-existent lock via REST

### PR DESCRIPTION
**Pull request by sven-urbanski-freiheit-com**

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer_test.go [hunk 1]
```diff
@@ -3777,7 +3773,7 @@ func setupRepositoryTest(t *testing.T) Repository {
 
 func setupRepositoryTestWithPath(t *testing.T) (Repository, string) {
 	dir := t.TempDir()
-	t.Logf("created dir %s", dir)
+	//t.Logf("created dir %s", dir)
```
Comment (bjoern-michaelsen-freiheit) [inline services/cd-service/pkg/repository/transformer_test.go:3776]: remove?

File: services/cd-service/pkg/grpc/errors.go [hunk 2]
```diff
@@ -37,11 +37,15 @@ func PublicError(ctx context.Context, err error) error {
...
 	return status.Error(codes.Canceled, err.Error())
 }
 
-func AuthError(ctx context.Context, err error) error {
+func FailedPrecondition(_ context.Context, err error) error {
+	return status.Error(codes.FailedPrecondition, err.Error())
+}
+
+func AuthError(_ context.Context, err error) error {
 	return status.Error(codes.Unauthenticated, "error: "+err.Error())
```
Comment (bjoern-michaelsen-freiheit) [inline services/cd-service/pkg/grpc/errors.go:49]: inconsistent with line 45/41, can we fix that?

**Review by bjoern-michaelsen-freiheit — APPROVED** [review]
approved with comments

File: services/cd-service/pkg/grpc/errors.go (hunk 2 above)
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/grpc/errors.go:49]: done

**Pull request merged**